### PR TITLE
[fix #251] test for identical username on changeUsername

### LINKED ISSUE #251: Catch rename to self
Issue description here: https://github.com/hoodiehq/hoodie-plugin-users/pull/18

### PR DESCRIPTION
**Pull request by lewiscowper**
Hopefully fixes #251.

### CODE REVIEW

**Comment by svnlto** [comment]
w00t!

**Comment by gr2m** [comment]
Looks good!

File: src/hoodie/account.js [hunk 1]
```diff
@@ -501,8 +501,14 @@ function hoodieAccount(hoodie) {
   // But the current password is needed to login with the new username.
   //
   account.changeUsername = function changeUsername(currentPassword, newUsername) {
-    newUsername = newUsername || '';
-    return changeUsernameAndPassword(currentPassword, newUsername.toLowerCase());
+    if (newUsername !== account.username) {
+      newUsername = newUsername || '';
```
Comment (janl) [inline src/hoodie/account.js:505]: I know it is from the old version, but this looks fishy, why should we set the username to `''`?

Comment (gr2m) [inline src/hoodie/account.js:505]: so that `newUsername.toLowerCase()` doesn't blow up. It will fail anyway when `newUsername` is falsy

Comment (janl) [inline src/hoodie/account.js:505]: Got it. Might be worth flagging for a rewrite. The intent of the code is not very clear.

I’d do something like this:

``` js
if (!newUsername) { // catches '' (empty string), false, null, undefined
  return false;
}
var lowerNewUsername = newUsernamme.toLowerCase();
return changeUsernameAndPassword(currentPassword, lowerNewUsername);
```

Comment (janl) [inline src/hoodie/account.js:505]: Thanks for the explanation! :)

**Pull request merged**